### PR TITLE
Schedule3: Events grouping within the selected date

### PR DESCRIPTION
**Pull request by yurii-lubynets**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/999

### CODE REVIEW

File: src/components/molecules/ScheduleEventListNG/ScheduleEventSubListNG.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+export const ScheduleEventSubListNG: React.FC<ScheduleEventSubListNGProps> = ({
+  events,
+  title,
+}) => {
+  const [showMoreEvents, setShowMoreEvents] = useState(false);
+  const toggleShowMoreEvents = useCallback(() => {
+    setShowMoreEvents(!showMoreEvents);
+  }, [showMoreEvents, setShowMoreEvents]);
```
Comment (sunny-viktoryia) [inline src/components/molecules/ScheduleEventListNG/ScheduleEventSubListNG.tsx:23]: let's use `useShowHide` hook here instead.

File: src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  const liveEvents = useMemo(() => daysEvents.filter(isEventLive), [
+    daysEvents,
+  ]);
+  const comingSoonEvents = useMemo(() => daysEvents.filter(isEventSoon), [
+    daysEvents,
+  ]);
+  const laterEvents = useMemo(() => daysEvents.filter(isEventLater), [
+    daysEvents,
+  ]);
```
Comment (sunny-viktoryia) [inline src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx:33]: when these constants will be re-calculated? I believe we should use some setInterval to check if the event moved from status "coming soon" to "live" (for example).

File: src/components/molecules/ScheduleEventListNG/utils.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+export const isEventLater = (event: PersonalizedVenueEvent) =>
+  !(
+    isEventLive(event) ||
+    isEventStartingSoon(event, EVENT_STARTING_SOON_TIMEFRAME)
+  );
+
+export const isEventSoon = (event: PersonalizedVenueEvent) =>
+  !isEventLive(event) &&
+  isEventStartingSoon(event, EVENT_STARTING_SOON_TIMEFRAME);
```
Comment (sunny-viktoryia) [inline src/components/molecules/ScheduleEventListNG/utils.ts:15]: maybe would be better to move them to utils/events too? what do you think?

**Review by sunny-viktoryia — COMMENTED** [review]
I liked that! :) a few comments and then should be good to go.

File: src/components/molecules/ScheduleEventListNG/ScheduleEventSubListNG.tsx (hunk 1 above)
Comment (yurii-lubynets) [reply src/components/molecules/ScheduleEventListNG/ScheduleEventSubListNG.tsx]: fixed in c7e4c1a90c7f5f57663972cbdd08ab5d051cb2d7

File: src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx (hunk 2 above)
Comment (yurii-lubynets) [reply src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx:33]: They will be re-calculated every time `daysEvents` are changed. Please clarify what you mean about `setInterval`?

I assume `isEventLive`, `isEventSoon` and `isEventLater` is used `setInterval` under the hood

File: src/components/molecules/ScheduleEventListNG/utils.ts (hunk 3 above)
Comment (yurii-lubynets) [reply src/components/molecules/ScheduleEventListNG/utils.ts]: fixed in 91ce25851e7b49d5cc1dc28c004ef21d37727e92

File: src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx (hunk 2 above)
Comment (sunny-viktoryia) [reply src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx:32]: have you checked that event is moving from "coming soon" to "live" list automatically once it's started? 

My suggestion about setInterval was that we should check every minute/5 minutes what's the status of events to move them to the appropriate group, and if the event is finished, hide it. does that make sense?

Comment (yurii-lubynets) [reply src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.tsx]: Got it. fixed in f3056b41c3cf3139daa66a9d68ca53e48c33f236

File: src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx [hunk 4]
```diff
@@ -142,7 +142,7 @@ export const ScheduleItemNG: React.FC<ScheduleItemNGProps> = ({ event }) => {
         <div className="ScheduleItemNG__name">{event.name}</div>
         <div className="ScheduleItemNG__place">
           <span className="ScheduleItemNG__place--location">in</span>{" "}
-          {event.room}
+          {event.room || eventVenue?.name}
```
Comment (sunny-viktoryia) [inline src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx:145]: thank you! 😍

**Review by sunny-viktoryia — APPROVED** [review]
looks good. 🙃 thanks for the ui fixes!

**Pull request merged**